internal/installer/install.go

```diff
@@ -93,13 +93,34 @@ func (id *installData) download() (string, error) {
 			url, res.Status,
 		)
 	}
+	resHash := res.Header.Get("X-Checksum-Sha256")
+
+	res, err = makeRequest(http.MethodGet, url)
+	if err != nil {
+		return "", err
+	}
+	defer res.Body.Close()
+
+	buf, err := io.ReadAll(res.Body)
```

Comment:
I'm reading the current process as:
* read res.Body fully
* check hash of body against known hash
* copy body to file

Since `tmp` always gets deleted, it might be simpler to use an `io.MultiWriter`:

```
hash := sha256.New()
if _, err := io.Copy(io.MultiWriter(tmp, hash), res.Body); err != nil {
    // ...
}

gotHash := hex.EncodeToString(hash.Sum(nil))
if want != gotHash {
    // err
}
```

This moves some of the hashing logic into the download func, which I think is fine--it's simple enough, and download is being tested in a way where if we break hashing, tests will fail. That might also let us clean up/remove util.go and util_test.go.